coordinator/src/db/positions.rs

```diff
@@ -0,0 +1,180 @@
+use crate::orderbook::db::custom_types::Direction;
+use crate::schema::positions;
+use crate::schema::sql_types::ContractSymbolType;
+use crate::schema::sql_types::PositionStateType;
+use anyhow::bail;
+use anyhow::Result;
+use diesel::prelude::*;
+use diesel::query_builder::QueryId;
+use diesel::result::QueryResult;
+use diesel::AsExpression;
+use diesel::FromSqlRow;
+use std::any::TypeId;
+use time::OffsetDateTime;
+
+#[derive(Queryable, Debug, Clone)]
+pub(crate) struct Position {
+    pub id: i32,
+    pub contract_symbol: ContractSymbol,
+    pub leverage: f32,
+    pub quantity: f32,
+    pub direction: Direction,
+    pub average_entry_price: f32,
+    pub liquidation_price: f32,
+    pub position_state: PositionState,
+    pub collateral: i64,
+    pub creation_timestamp: OffsetDateTime,
+    pub expiry_timestamp: OffsetDateTime,
+    pub update_timestamp: OffsetDateTime,
+    pub trader: String,
+}
+
+impl Position {
+    /// Returns the position by id
+    pub fn get_open_position_by_trader(
+        conn: &mut PgConnection,
+        trader: String,
```

Comment:
```suggestion
        trader_pubkey: String,
```

It would be great to specify (in comment or better variable name) what this `String` represents. The pubkey?